superset-frontend/src/components/ReportModal/index.tsx

```diff
@@ -135,31 +146,29 @@ const ReportModal: FunctionComponent<ReportProps> = ({
     setCurrentReport({ type, payload });
   }, []);
   const [error, setError] = useState<CronError>();
-  // ---------- comments on lines 21, 28, 125, 139-159 & 182 are being held for edit functionality
-  // const [hasConnectedReport, setHasConnectedReport] = useState<boolean>(false);
   // const [isLoading, setLoading] = useState<boolean>(false);
   const dispatch = useDispatch();
-
   // Report fetch logic
-  // const {
-  //   state: { resource },
-  // } = useSingleViewResource<ReportObject>(
-  //   'report',
-  //   t('report'),
-  //   addDangerToast,
-  // );
-
-  // useEffect(() => {
-  //   if (resource?.dashboard) {
-  //     setHasConnectedReport(true);
-  //   }
-  // }, [resource?.dashboard]);
-
+  const reports = useSelector<any, AlertObject>(state => state.reports);
+  const isEditMode = reports && Object.keys(reports).length;
```

Comment:
nit, but you can also do `const isEditMode = Object.keys(reports || []).length;`